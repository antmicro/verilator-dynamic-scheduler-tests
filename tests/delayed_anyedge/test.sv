module delay_anyedge ();
    reg flag_a;
    reg flag_b;
    logic clk = 1;
    initial
      forever
      begin
        #20;
        clk = ~clk;
      end
    always @(posedge clk)
    begin
        flag_b <= 1'b0;
        #5;
        flag_a <= 1'b1;
        #10;
        flag_b <= 1'b1;
    end
    always @(flag_a)
    begin
        if (flag_b !== 1'b0) $stop;
        #25;
        if (flag_b !== 1'b1) $stop;
        $finish;
    end
endmodule
