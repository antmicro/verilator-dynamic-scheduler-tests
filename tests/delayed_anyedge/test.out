Will wait for the event here...
Will trigger the event in #100
triggering!
Got the event!
*-* All Finished *-*
